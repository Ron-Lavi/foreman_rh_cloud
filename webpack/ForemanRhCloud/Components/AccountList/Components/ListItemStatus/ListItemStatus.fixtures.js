export const props = {
  account: {
    generate_report_status: 'Exit Code: pid 22191 exit 0',
    upload_report_status: 'Exit Code: Running in pid 22376',
    label: 'Default Org',
  },
};
