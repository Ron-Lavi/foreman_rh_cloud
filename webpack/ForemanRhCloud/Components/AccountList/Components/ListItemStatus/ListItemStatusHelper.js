import React from 'react';
import { Icon, Spinner } from 'patternfly-react';

const STATUS_ICONS = {
  success: <Icon name="check" />,
  failure: <Icon name="times" />,
  running: <Spinner loading inline size="xs" />,
  unknown: <span>--</span>,
};

export const getStatusIconByRegex = status => {
  if (!status || status === 'unknown') {
    return STATUS_ICONS.unknown;
  }

  const statusCopy = status.toLowerCase();
  if (statusCopy.indexOf('exit 0') !== -1) {
    return STATUS_ICONS.success;
  }

  if (
    statusCopy.indexOf('running') !== -1 ||
    statusCopy.indexOf('restarting') !== -1
  ) {
    return STATUS_ICONS.running;
  }

  return STATUS_ICONS.failure;
};
