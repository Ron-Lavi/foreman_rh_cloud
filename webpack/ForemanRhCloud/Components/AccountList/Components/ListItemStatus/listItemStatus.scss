.status {
  max-width: 250px;

  .item {
    text-align: center;
    font-weight: 600;

    p {
      margin: auto;
    }
  }

  .spinner {
    float: none;
  }
}
