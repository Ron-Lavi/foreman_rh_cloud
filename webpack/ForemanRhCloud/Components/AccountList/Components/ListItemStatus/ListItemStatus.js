import React from 'react';
import { Grid } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import './listItemStatus.scss';
import { getStatusIconByRegex } from './ListItemStatusHelper';

const ListItemStatus = ({ account }) => {
  const generatingStatusIcon = getStatusIconByRegex(
    account.generate_report_status
  );
  const uploadingStatusIcon = getStatusIconByRegex(
    account.upload_report_status
  );
  return (
    <Grid className="status">
      <Grid.Col sm={6} className="item">
        <p>{__('Generating')}</p>
        {generatingStatusIcon}
      </Grid.Col>
      <Grid.Col sm={6} className="item">
        <p>{__('Uploading')}</p>
        {uploadingStatusIcon}
      </Grid.Col>
    </Grid>
  );
};

ListItemStatus.propTypes = {
  account: PropTypes.shape({
    generate_report_status: PropTypes.string,
    upload_report_status: PropTypes.string,
  }),
};

ListItemStatus.defaultProps = {
  account: {
    generate_report_status: 'unknown',
    uploupload_report_statusading: 'unknown',
  },
};

export default ListItemStatus;
