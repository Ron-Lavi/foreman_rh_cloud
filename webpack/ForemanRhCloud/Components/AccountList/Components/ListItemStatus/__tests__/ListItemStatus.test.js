import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import ListItemStatus from '../ListItemStatus';
import { props } from '../ListItemStatus.fixtures';

const fixtures = {
  'render without Props': {},
  'render with Props': props,
};

describe('ListItemStatus', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ListItemStatus, fixtures));
});
