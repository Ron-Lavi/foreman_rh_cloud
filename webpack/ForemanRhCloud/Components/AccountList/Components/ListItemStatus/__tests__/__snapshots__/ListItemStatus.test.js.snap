// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemStatus rendering render with Props 1`] = `
<Grid
  bsClass="container"
  className="status"
  componentClass="div"
  fluid={false}
>
  <Col
    bsClass="col"
    className="item"
    componentClass="div"
    sm={6}
  >
    <p>
      Generating
    </p>
    <Icon
      name="check"
      type="fa"
    />
  </Col>
  <Col
    bsClass="col"
    className="item"
    componentClass="div"
    sm={6}
  >
    <p>
      Uploading
    </p>
    <Spinner
      className=""
      inline={true}
      inverse={false}
      loading={true}
      size="xs"
    />
  </Col>
</Grid>
`;

exports[`ListItemStatus rendering render without Props 1`] = `
<Grid
  bsClass="container"
  className="status"
  componentClass="div"
  fluid={false}
>
  <Col
    bsClass="col"
    className="item"
    componentClass="div"
    sm={6}
  >
    <p>
      Generating
    </p>
    <span>
      --
    </span>
  </Col>
  <Col
    bsClass="col"
    className="item"
    componentClass="div"
    sm={6}
  >
    <p>
      Uploading
    </p>
    <span>
      --
    </span>
  </Col>
</Grid>
`;
